### PR TITLE
Update Docker image tag creation

### PR DESCRIPTION
**Pull request by sysrex**
Update Docker image tag creation
- Generate sha tag for main branch
- generate branch name tag for every other branch

### CODE REVIEW

File: .github/workflows/docker-build.yml [hunk 1]
```diff
@@ -29,7 +29,8 @@ jobs:
         with:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
           tags: |
-            type=sha
+            type=sha,format=long,enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
+            type=ref,enable=${{ github.ref != format('refs/heads/{0}', 'main') }},event=branch
```
Comment (liamsi) [inline .github/workflows/docker-build.yml:33]: How will the resulting tag look like?

Comment (sysrex) [reply .github/workflows/docker-build.yml:33]: if the branch is main the docker tag will be sha-xxxxxxx, for any other branch the docker tag is the branch name (that way we can create a matrix of testing regardless of where the docker image is coming from and we're also able to differentiate between them)

Comment (sysrex) [reply .github/workflows/docker-build.yml:33]: yes, that is correct

Comment (Wondertan) [reply .github/workflows/docker-build.yml:33]: I am fine with this, but don't understand the usecase behind this, can you elaborate?

**Comment by Wondertan** [comment]
@sysrex, how should we proceed here?

**Comment by Wondertan** [comment]
Do we still need this?

**Comment by Bidon15** [comment]
bumping this @sysrex

**Review by SweeXordious — COMMENTED** [review]
Awesome stuff. I guess it makes sense also to do the same for celestia-app

**Comment by sysrex** [comment]
We still require this to have the docker image tag match the commit sha, this is imperative to be able to continue unobstructed working on the devnet.

**Pull request closed without merging**